gocourse02/main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"fmt"
+	"math/rand/v2"
+)
+
+/*
+Написати програму “Зоопарк”. Звіри повтікали (більше трьох штук), наглядач повинен їх зібрати.
+Кожна сутність (наглядач, звір, клітка, …) представляється окремою структурою (zookeeper, animal, cage, …).
+Треба використати щонайменше: структури, вказівник, nil, будування, конструктор. Додати тваринам можливість
+розмножуватись. Програма має демонструвати свою роботу через вивід в stdout.
+*/
+
+const escapedAnimalsCount = 15 // Кількість звірів які повтікали
+
+type ZooKeeper struct {
+	Name          string
+	AnimalsFound  int
+	AnimalsCaught int
+}
+
+type Catcher interface {
+	FindEscapedAnimal() bool                // Спроба знайти звіра
+	AttemptCatchAnimal(*Cage, *Animal) bool // Сброба зловити звіра в клітку
+}
+
+type Animal struct {
+	ID     int
+	Name   string
+	Gender string
+	Weight int
+}
+
+type Cage struct {
+	MaxSeats      int
+	MaxWeight     int
+	CurrentSeats  int
+	CurrentWeight int
+	Animal        []Animal
```

Comment:
Рекомендація перейменувати поле: додати суфікс `s` до `Animal`, тому що це поле, де може бути багато звірів.

```suggestion
	Animals       []Animal
```

Я не знайшов формального правила, за яким обов'язково треба додавати `s` до імен слайсів. Але так з мого досвіду всі роблять.

Як аргумент, можу привести код Go для змінних типу слайс стрінгів `[]string` https://github.com/search?q=repo%3Agolang%2Fgo%20%5B%5Dstring%7B&type=code

З пошуку видно, що назви такі:

```
typeFiles
mimeGlobs
platformZoneSources
IllegalPrefixes
logTypes
logPaths
```

Але є й такі:
```
skip
htmlReplacementTable
list
ss
Register
```

Що є виключенням треба дивитись конкретний контекст, чому так назвали.